src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -423,18 +430,43 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     }
 
     fun addSuperinterfaces(superinterfaces: Iterable<TypeName>) = apply {
-      this.superinterfaces += superinterfaces
+      this.superinterfaces.putAll(superinterfaces.map {
+        Pair(it, null) 
+      })
     }
 
-    fun addSuperinterface(superinterface: TypeName) = apply {
-      superinterfaces += superinterface
+    fun addSuperinterface(superinterface: TypeName, delegate: CodeBlock = CodeBlock.EMPTY) = apply {
+      if (delegate.isEmpty()) {
+        this.superinterfaces.put(superinterface, null)
```

Comment:
Should we use `CodeBlock.EMPTY` instead of null in the map?

Reply:
(seems simpler to have a single value that means “no delegate” rather than two)

Reply:
Actually that might be cleaner. I'll push formatting fix but wait for other feedback on this (since I've already re-done a few times) - would it be possible to do something more specific than just `CodeBlock`s and nulls?

Reply:
something that could have standard lib delegate & operator function support too?

Reply:
@swankjesse I remember we've discussed sentinels on an earlier PR. I think `null`s play better in Kotlin (might not be true for Java) as the compiler ensures you don't miss the checks. If your sentinel is `CodeBlock.EMPTY` you have to be careful to check against it everywhere, and it feels super error-prone.

Reply:
Agreed in general. In this case the code is already actively checking if the code block is empty and mapping that to null. It seems bad to use empty code blocks as a sentinel in the API but nulls in the implementation.

I don't feel strongly here and to unblock this, let's keep it as is!